vm/actor/src/builtin/market/mod.rs

```diff
@@ -863,6 +864,13 @@ impl Actor {
                             "deal should not be slashed and should have a schedule for next cron"
                         );
```

Comment:
Can't comment on these lines, but the assertions should be removed (you're checking this in the lines you added below). I might have missed some above, but be sure to remove all those assertions